src/commands/calendar/winner.ts

```diff
@@ -0,0 +1,75 @@
+import { SlashCommandBuilder } from "@discordjs/builders";
+import { AudioPlayerStatus, VoiceConnectionStatus } from "@discordjs/voice";
+import { CommandInteraction, GuildMember, Message, MessageEmbed, VoiceChannel } from "discord.js";
+import path from "path";
+
+import { DiscordClient, SlashCommandHandler } from "../../client";
+import logger from "../../utils/logger";
+import { createVoiceChannelConnection, playSong, randomNumber } from "../../utils/utils";
+
+export default class WinnerCommand extends SlashCommandHandler {
+  builder = new SlashCommandBuilder()
+    .setName("calendar")
+    .setDescription("Choose a random member of your voice channel to win!"); //.setAlias("kalender")
+
+  async handle(interaction: CommandInteraction, client: DiscordClient): Promise<void> {
+    if (!interaction.guild) return;
+
+    if (interaction.member instanceof GuildMember) {
+      const voiceChannel = interaction.member.voice.channel;
+      if (voiceChannel && voiceChannel instanceof VoiceChannel) {
+        const members = Array.from(voiceChannel.members.values());
+
+        if (members.length === 1) {
+          return await interaction.reply("Trying to win on your own, eh? You lost.");
+        }
+
+        const number = await randomNumber(0, members.length - 1);
+        const winner = members[number];
+
+        // Wait with reply so we can send result and audio at the same time
+        await interaction.deferReply();
+
+        // Play audio:
+        try {
+          await playSong(client.audioPlayer, path.resolve(process.cwd(), "assets/restricted/hes_the_winner.mp3"));
+
+          const connection = await createVoiceChannelConnection(voiceChannel);
+          connection.subscribe(client.audioPlayer);
+
+          client.audioPlayer.on(AudioPlayerStatus.Idle, () => {
+            client.audioPlayer.stop();
+            if (connection && connection.state.status != VoiceConnectionStatus.Destroyed) connection.destroy();
+          });
+        } catch (err) {
+          logger.error(err);
+        }
+
+        // Fancy message embed:
+        const embed = new MessageEmbed()
+          .setColor("#0099ff")
+          .setTitle("The Winner")
+          .setDescription(`<@${winner.user.id}>`)
+          .setThumbnail(
+            winner.user.avatarURL() ?? "https://www.dictionary.com/e/wp-content/uploads/2018/03/PogChamp-300x300.jpg",
```

Comment:
Ville kanskje lagt med bildet i repoet her og linket direkte til det derfra, typ lastet det opp i `assets` og lenket til det som https://raw.githubusercontent.com/lesesalen/lesebot/master/assets/blobross.png f.eks., for å unngå at lenket plutselig forsvinner.